### PR TITLE
Removed legacy ORM lookup support per deprecation timeline; refs #16187.

### PR DESCRIPTION
**Pull request by timgraham**
deprecated in 20bab2cf9

### CODE REVIEW

**Comment by timgraham** [comment]
Anssi, could you verify this? Possibly there are more cleanups that could be done, I'm not sure.

There are still some references to `Constraint` in comments that may need to be fixed up.
https://github.com/django/django/blob/bd98926f0eb19d27821a8a7679b42ff46e53e4da/django/db/models/sql/where.py#L41
https://github.com/django/django/blob/bd98926f0eb19d27821a8a7679b42ff46e53e4da/django/db/models/sql/where.py#L56
https://github.com/django/django/blob/bd98926f0eb19d27821a8a7679b42ff46e53e4da/django/db/models/sql/where.py#L291

**Comment by akaariai** [comment]
Looks fine to me except for the one comment I added.

Do we have coverage builds currently? If not, then running at least a subset of tests and checking which parts of the ORM do not get called could reveal more deprecated or dead code.

File: django/db/models/sql/where.py [hunk 1]
```diff
@@ -160,109 +150,10 @@ def get_group_by_cols(self):
             if hasattr(child, 'get_group_by_cols'):
                 cols.extend(child.get_group_by_cols())
             else:
-                if isinstance(child[0], Constraint):
-                    cols.append((child[0].alias, child[0].col))
                 if hasattr(child[3], 'get_group_by_cols'):
```
Comment (akaariai) [inline django/db/models/sql/where.py]: This looks strange to me. I'll check why the child[3] check is here. If there is a good reason, then lets add a comment.

**Comment by timgraham** [comment]
Coverage build (for SQLite) is here: http://djangoci.com/job/django-coverage/HTML_Coverage_Report/
We can setup some builds for other databases if that would be helpful.

I did some additional removals in `where.py` based on that -- you could probably do some more cleanup based on the coverage report when you have time, but this seems to be the majority of the deprecation related code as far as I can tell.

Could you propose an update to this comment in the PR description about `Constraint` that I could merge with this? https://github.com/django/django/blob/bd98926f0eb19d27821a8a7679b42ff46e53e4da/django/db/models/sql/where.py#L41

**Comment by akaariai** [comment]
I think the proper way to word that is to mention that a child is an expression producing Boolean values. Most likely the expression is a Lookup instance, but other types of objects fulfilling the required API could be used, too (for example, sql.where.EverythingNode).

Currently it seems only as_sql() is required, but I am not sure if that is still true after we have done full cleanup for the deprecation code (for example, get_group_by_cols() seems to be required, too).

**Pull request merged**